sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs

```diff
@@ -44,7 +51,7 @@ public override ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpP
 
         private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool async)
         {
-            int attempt = 0;
+            message.PipelineContext.RetryAttempt = 0;
```

Comment:
Why are we setting this value here to 0? I think the ctor for the context should set it to zero and then the retry policy should be only incrementing it.